rust/build.rs

```diff
@@ -0,0 +1,17 @@
+use std::path::Path;
+
+fn main() {
+    let dir = Path::new("src");
+
+    cc::Build::new()
+        .include(&dir)
+        .file(dir.join("parser.c"))
+        .flag_if_supported("-Wno-unused-but-set-variable")
+        .compile("tree-sitter-parser");
+
+    cc::Build::new()
+        .cpp(true)
+        .include(&dir)
+        .file(dir.join("scanner.cc"))
+        .compile("tree-sitter-scanner");
```

Comment:
We need two separate build commands because I cannot get `g++` to build `parser.c` (`gcc` can do it).

Reply:
Interesting `clang++` will build `parser.c` (although generates a lot of warnings).